client/my-sites/pages/page-list.jsx

```diff
@@ -85,7 +85,7 @@ export default class PageList extends Component {
 		return (
 			<div>
 				<QueryPosts siteId={ siteId } query={ query } />
-				<QuerySiteSettings siteId={ siteId } />
+				{ siteId && <QuerySiteFrontPage siteId={ siteId } /> }
```

Comment:
We can check for siteId in the QueryComponent. If it's not set it should be easy to no-op to avoid the caller from doing { siteId && <Query }